src/crafting.cpp

```diff
@@ -1102,14 +1102,45 @@ comp_selection<item_comp> player::select_item_component( const std::vector<item_
         uimenu cmenu;
         // Populate options with the names of the items
         for( auto &map_ha : map_has ) {
-            std::string tmpStr = item::nname( map_ha.type ) + _( " (nearby)" );
+            std::string tmpStr;
+            if( item::count_by_charges( map_ha.type ) ) {
+                tmpStr =  string_format( _( "%s (%d charges nearby)" ),
+                                        item::nname( map_ha.type ).c_str(),
+                                        map_inv.charges_of( map_ha.type ));
+            }
+            else {
+                tmpStr = string_format( _( "%s (%d nearby)" ),
+                                       item::nname( map_ha.type ).c_str(),
+                                       map_inv.amount_of( map_ha.type ) );
+            }
             cmenu.addentry( tmpStr );
         }
         for( auto &player_ha : player_has ) {
-            cmenu.addentry( item::nname( player_ha.type ) );
+            std::string tmpStr;
+            if( item::count_by_charges( player_ha.type ) ) {
+                tmpStr =  string_format( _( "%s (%d charges on person)" ),
+                                        item::nname( player_ha.type ).c_str(),
+                                        charges_of( player_ha.type ));
```

Comment:
Does that include the ones after a '('? I could not understand most of doc/CODE_STYLE.txt.

For example, is it c_str( ) or c_str()?